### ISSUE TITLE
Problem with (Object[]) new String[]{"foo"}

### ISSUE THREAD

**Issue by bobjacobsen** [spotbugs/spotbugs#758]
This construct passes as an Object[] method parameter 
```
(Object[]) new String[]{"foo"}
```
causes SpotBugs 3.1.7 to throw numerous exceptions.  (Yes, it's not great code, but it compiles) A typical stack trace follows.  

```
 [spotbugs]   Error processing opcode checkcast @ 31 in jmri.jmrix.pricom.pockettester.MonitorFrame.title : ()Ljava.lang.String;
 [spotbugs]     java.lang.IllegalArgumentException: Unknown signature [Ljava/lang/Object; for number 1
 [spotbugs]       At edu.umd.cs.findbugs.OpcodeStack$Item.<init>(OpcodeStack.java:671)
 [spotbugs]       At edu.umd.cs.findbugs.OpcodeStack.sawOpcode(OpcodeStack.java:1725)
 [spotbugs]       At edu.umd.cs.findbugs.bcel.OpcodeStackDetector.afterOpcode(OpcodeStackDetector.java:83)
 [spotbugs]       At edu.umd.cs.findbugs.visitclass.DismantleBytecode.visit(DismantleBytecode.java:884)
 [spotbugs]       At edu.umd.cs.findbugs.detect.FieldItemSummary.visit(FieldItemSummary.java:130)
 [spotbugs]       At edu.umd.cs.findbugs.visitclass.BetterVisitor.visitCode(BetterVisitor.java:218)
 [spotbugs]       At edu.umd.cs.findbugs.visitclass.PreorderVisitor.visitCode(PreorderVisitor.java:243)
 [spotbugs]       At edu.umd.cs.findbugs.bcel.OpcodeStackDetector.visitCode(OpcodeStackDetector.java:65)
 [spotbugs]       At org.apache.bcel.classfile.Code.accept(Code.java:132)
 [spotbugs]       At edu.umd.cs.findbugs.visitclass.PreorderVisitor.doVisitMethod(PreorderVisitor.java:315)
 [spotbugs]       At edu.umd.cs.findbugs.visitclass.PreorderVisitor.visitJavaClass(PreorderVisitor.java:403)
 [spotbugs]       At org.apache.bcel.classfile.JavaClass.accept(JavaClass.java:213)
 [spotbugs]       At edu.umd.cs.findbugs.BytecodeScanningDetector.visitClassContext(BytecodeScanningDetector.java:38)
 [spotbugs]       At edu.umd.cs.findbugs.DetectorToDetector2Adapter.visitClass(DetectorToDetector2Adapter.java:76)
 [spotbugs]       At edu.umd.cs.findbugs.FindBugs2.analyzeApplication(FindBugs2.java:1080)
 [spotbugs]       At edu.umd.cs.findbugs.FindBugs2.execute(FindBugs2.java:281)
 [spotbugs]       At edu.umd.cs.findbugs.FindBugs.runMain(FindBugs.java:401)
 [spotbugs]       At edu.umd.cs.findbugs.FindBugs2.main(FindBugs2.java:1185)
```

**Comment by KengoTODA** [comment]
related with #688

**Comment by wreulicke** [comment]
I have same issue.

I reproduced this issue.

repository: https://github.com/wreulicke/spotbugs-758-reproduce
CI: https://circleci.com/gh/wreulicke/spotbugs-758-reproduce/1

**Comment by iloveeclipse** [comment]
@bobjacobsen : please can you attach simple test case to reproduce the issue? It would be nice if the fix in #760 would have a regression test.
